File: decompiled/sources/com/streamax/client/SingleViewInfo.java
package com.streamax.client;

import com.dvr.net.DvrNet;

public class SingleViewInfo {
    public static final int CONN_FAILURE = 1;
    public static final int CONN_SUCCESS = 0;
    public int mChannel;
    public int mConnState = 0;
    public DevInfoBean mDevInfo;
    public String mDevIp;
    private DvrNet mDvrNet;

    public SingleViewInfo() {
        reset();
    }

    public void setConnState(int i) {
        this.mConnState = i;
    }

    public int getConnState() {
        return this.mConnState;
    }

    public void reset() {
        this.mDvrNet = null;
        this.mDevInfo = new DevInfoBean();
        this.mChannel = -1;
    }

    public boolean check(DevInfoBean devInfoBean, int i) {
        return i != -1 && devInfoBean.mDevIp.equals(this.mDevInfo.mDevIp) && devInfoBean.mMediaPort == this.mDevInfo.mMediaPort && i == this.mChannel;
    }

    public boolean check(DevInfoBean devInfoBean) {
        if (this.mChannel != -1 && devInfoBean.mDevIp.equals(this.mDevInfo.mDevIp) && devInfoBean.mMediaPort == this.mDevInfo.mMediaPort) {
            return true;
        }
        return false;
    }

    public DevInfoBean getDeviceInfoBean() {
        return this.mDevInfo;
    }

    public void setDeviceInfo(DevInfoBean devInfoBean) {
        this.mDevInfo = devInfoBean;
    }

    public DvrNet getNet() {
        return this.mDvrNet;
    }

    public void setNet(DvrNet dvrNet) {
        this.mDvrNet = dvrNet;
    }
}

File: decompiled/sources/com/streamax/client/SingleViewInfo.kt
package com.streamax.client

import com.dvr.net.DvrNet

class SingleViewInfo {
    @JvmField var mChannel: Int = -1
    @JvmField var mConnState: Int = 0
    @JvmField var mDevInfo: DevInfoBean = DevInfoBean()
    @JvmField var mDevIp: String? = null
    private var mDvrNet: DvrNet? = null

    fun setConnState(state: Int) { mConnState = state }
    fun getConnState(): Int = mConnState
    fun reset() {
        mDvrNet = null
        mDevInfo = DevInfoBean()
        mChannel = -1
    }
    fun check(devInfoBean: DevInfoBean, channel: Int): Boolean =
        channel != -1 && devInfoBean.mDevIp == mDevInfo.mDevIp &&
            devInfoBean.mMediaPort == mDevInfo.mMediaPort && channel == mChannel
    fun check(devInfoBean: DevInfoBean): Boolean =
        mChannel != -1 && devInfoBean.mDevIp == mDevInfo.mDevIp &&
            devInfoBean.mMediaPort == mDevInfo.mMediaPort
    fun getDeviceInfoBean(): DevInfoBean = mDevInfo
    fun setDeviceInfo(info: DevInfoBean) { mDevInfo = info }
    fun getNet(): DvrNet? = mDvrNet
    fun setNet(net: DvrNet?) { mDvrNet = net }

    companion object {
        const val CONN_FAILURE: Int = 1
        const val CONN_SUCCESS: Int = 0
    }
}
